### ISSUE TITLE
mediaSettingsRange should return step

### ISSUE THREAD

**Issue by beaufortfrancois** [w3c/mediacapture-image#105]
It should be very easy for developers to specify the `step` value in `<input type="range">` HTML elements.
Please provide this information when available.

```diff
        interface MediaSettingsRange {
            readonly attribute long max;
            readonly attribute long min;
            readonly attribute long current;
+           readonly attribute long step;
        };
```

For info, I can get it with a Logitech C920:

```sh
$ v4l2-ctl --dev=/dev/video1 -L        
                     brightness (int)    : min=0 max=255 step=1 default=128 value=128
                       contrast (int)    : min=0 max=255 step=1 default=32 value=32
                     saturation (int)    : min=0 max=255 step=1 default=32 value=32
 white_balance_temperature_auto (bool)   : default=1 value=1
                           gain (int)    : min=0 max=255 step=1 default=64 value=59
           power_line_frequency (menu)   : min=0 max=2 default=2 value=1
                                0: Disabled
                                1: 50 Hz
                                2: 60 Hz
      white_balance_temperature (int)    : min=2800 max=6500 step=1 default=4000 value=4428 flags=inactive
                      sharpness (int)    : min=0 max=255 step=1 default=72 value=72
         backlight_compensation (int)    : min=0 max=1 step=1 default=0 value=0
                  exposure_auto (menu)   : min=0 max=3 default=3 value=3
                                1: Manual Mode
                                3: Aperture Priority Mode
              exposure_absolute (int)    : min=3 max=2047 step=1 default=166 value=200 flags=inactive
         exposure_auto_priority (bool)   : default=0 value=1
                   pan_absolute (int)    : min=-36000 max=36000 step=3600 default=0 value=0
                  tilt_absolute (int)    : min=-36000 max=36000 step=3600 default=0 value=0
                 focus_absolute (int)    : min=0 max=255 step=17 default=68 value=85 flags=inactive
                     focus_auto (bool)   : default=1 value=1
                  zoom_absolute (int)    : min=1 max=5 step=1 default=1 value=5

**Comment by gmandyam** [comment]
@beaufortfrancois 

The original thinking was that the user agent would exercise best effort in meeting the setting requested by the developer , but it wouldn't have to to expose step to the developer.  This was intended to minimize the information that a website can use to fingerprint the device. 

See https://www.w3.org/TR/fingerprinting-guidance/#bp-summary for more information.

**Comment by beaufortfrancois** [comment]
Having already access to `min`, `current`, and `max` values is enough for fingerprinting I believe.
My hope is to help web developers build great UX with this valuable setting.

**Comment by gmandyam** [comment]
@beaufortfrancois 
>Having already access to min, current, and max values is enough for fingerprinting I believe.

I don't agree.  The User Agent can map min/current/max to a common scale across all cameras, so that it would be difficult for a website to identify a specific make/model of camera from these attributes alone.  An step attribute may be more difficult to abstract across cameras.

**Comment by beaufortfrancois** [comment]
Here's what I already have with the current situation:

```
MediaStreamTrack
  enabled: true
  id: "a77c882b-871c-4653-9163-128b1f44001d"
  kind: "video"
  label: "NCM-G102 (2232:6001)"

ImageCapture.getPhotoCapabilities()
  whiteBalanceMode: continuous
  colorTemperature: { min: 2800, current: 4000, max: 6500 }
  exposureMode: continuous
  exposureCompensation: { min: 0, current: 0, max: 0 }
  iso: { min: 0, current: 0, max: 0 }
  redEyeReduction: false
  focusMode: none
  brightness: { min: 0, current: 133, max: 255 }
  contrast: { min: 0, current: 255, max: 255 }
  saturation: { min: 0, current: 53, max: 100 }
  sharpness: { min: 0, current: 0, max: 7 }
  imageHeight: { min: 0, current: 0, max: 0 }
  imageWidth: { min: 0, current: 0, max: 0 }
  zoom: { min: 0, current: 0, max: 0 }
  fillLightMode: none
```

Do you think adding `step` is going to allow a website to identify a specific make/model of camera with everything that is already available?

**Comment by gmandyam** [comment]
@beaufortfrancois 

The example you cited showed the Logitech camera with step defined for several parameters.  Android camera API shows it only for one, exposureCompensation - see https://developer.android.com/reference/android/hardware/Camera.Parameters.html.  

So yes, I do think adding step can add fingerprinting surface that would allow identification of a camera that the existing parameters would not.

But I would like to get the co-editor's opinion.  @miguelao, what do you think?

**Comment by beaufortfrancois** [comment]
FWIW, [getZoomRatios](https://developer.android.com/reference/android/hardware/Camera.Parameters.html#getZoomRatios()) allows you too to access different "steps" of zoom in the Android Camera API.

Having a nice and consistent experience for web developers is my only wish there.

**Comment by miguelao** [comment]
@gmandyam, I felt @beaufortfrancois's pain when trying to map a `MediaSettingsRange` (e.g. zoom) to a slider element :smile:
IMHO adding |step| would not be a large modification, eases developers' lives and it's essentially provided in all platforms -- in Android, where not specifically mentioned, it's assumed to be one,  note that e.g. zoom in the old API [is a discrete set](https://developer.android.com/reference/android/hardware/Camera.Parameters.html#getZoomRatios()), so in this situation `step` would help.

**Comment by gmandyam** [comment]
@beaufortfrancois, @miguelao 

OK - let's go ahead and add.

**Comment by miguelao** [comment]
Closed per #109.

**Comment by miguelao** [comment]
FTR, I landed this in Cr and will probably be in tomorrow's Canary &ge; 56.0.2891.0 (probably)
